### ISSUE TITLE
Switches not triggering rules

### ISSUE THREAD

**Issue by jeremyquinn** [ebaauw/homebridge-hue#18]
I have been having my first play at replacing actions for Hue Tap and Dimmer switches, from being setup in the Hue Bridge (setup in iConnectHue) to using automation rules in HomeKIT.

The switches are not recognised by Apple's Home.app but are visible in Eve and Home (by Hochgatterer) which is where I tried this.

I created two scenes (specific lights ON & OFF) activated by two triggers (specific switch ON & OFF).
e.g. a Hue Tap's triggers were `Programmable Switch Output State → 1` for ON and `Programmable Switch Output State → 3` for OFF. (1 & 4 for the Dimmer)

The two scenes function normally.
The two triggers very seldom work, i.e. they are not actually triggered.

I can see the value changing in homebridge logs, eg.

`[11/16/2016, 8:59:01 PM] [Hue] Living Room Dimmer: set homekit output state from 1 to 4 on Wed Nov 16 2016 20:59:00` 

Only once or twice in testing did I see the logs of the scene running as a result of the trigger. The trigger typically saying it had never been triggered.

Should I be able to do this?
Is there anything I can do to help you debug this?

**Comment by ebaauw** [comment]
Hi @jeremyquinn 

Yes, you should be able to do this. I prefer to keep the rules concerning my lights in the hue bridge (for speed and for reliability), but I do operate my Eve Energy plugs and my Sonos speakers this way. (The Sonos speakers through another homebridge plugin that I haven't published yet).

I'm afraid this is an issue with homekit itself.  If the message appears in the log, homebridge-hue has recognised that you pressed the tap and sent the change to homekit.  You could double-check that the Output State value changes in Eve to be sure that the change has been received by homekit (tip: see the Types tab - use Edit to include Output State, Last Updated, etc.).

I assume you're running iOS 10 (as you mention the Apple Home app) and you have setup an AppleTV or iPad as homekit hub?  I reluctantly upgraded my AppleTV to a Gen 4 model, as my old Gen 3 model seemed to have a hard time coping with my homekit rules.

I find that homekit doesn't react well to changes to its rules (or scenes or room assignments).  Sometimes it ignores them, or even reverts them.  The changes are made to the homekit database on your iOS device and then sync'ed over iCloud with your other iOS devices and homekit hub.  This process is a bit flaky in my experience, and I suspect the rules don't end up correctly in the homekit hub.

What typically works for me to remedy rules not firing where they should:
- Disable all rules on one iOS device;
- Check that these changes are sync'ed over iCloud: your other devices should show all the rules as disabled as well.  Typically this takes multiple minutes...  As in many...
- Enable all rules on one iOS device.
- Check that your other devices show all the rules as enabled.   Unfortunately, there's no way to check the rules on your AppleTV, but if you have an iPad, you could (temporarily) set it up as homebridge hub.
- Now (hopefully) the rules fire.

Please let me know if this works for you.
Erik.

**Comment by jeremyquinn** [comment]
Thanks for your helpful response, Erik.
Yes, I see the output state change in Eve etc.
Yes, running iOS 10 and Gen 4 AppleTV.
I went through your procedure to re-sync the hub a few times. The rules maybe worked once or twice, then always began failing again. I did not try using an iPad as the hub.
I am not sure I have enough useful info to post a Radar about this, any suggestions?
Meanwhile I reluctantly reverted to configuring the switches on the Hue hub. You are right, it is faster.
Many thanks
Jeremy
